### PR TITLE
Single tenant auth, issuer url fix

### PR DESCRIPTION
**Pull request by muhammadmeisam**
## Info on Pull Request
The deploy.py is only able to set the issuer url to the redmond tenant in the case of single tenant auth. For multi tenant, since the the tenant_domain is provided specifically, it works fine. In the case of single tenant, if the instance is not in the redmond tenant, the cli is unable to authenticate itself.

_What does this include?_

## Validation Steps Performed
A test instance was deployed in the PME instance and it was verified that the cli could authenticate with the instance.

### CODE REVIEW

File: src/deployment/deploy.py [hunk 1]
```diff
@@ -427,6 +427,14 @@ def try_sp_create() -> None:
 
         else:
             authorize_application(uuid.UUID(ONEFUZZ_CLI_APP), app.app_id)
+            if self.multi_tenant_domain:
+                authority = COMMON_AUTHORITY
+            else:
+                authority = cli_app.authority
+            self.cli_config = {
+                "client_id": cli_app.client_id,
```
Comment (chkeita) [inline src/deployment/deploy.py:435]: is this value different from `ONEFUZZ_CLI_AUTHORITY` when deploying outside of the redmond instance ?
Can you please share the value you see

Comment (muhammadmeisam) [reply src/deployment/deploy.py:435]: 72f988bf-86f1-41af-91ab-2d7cd011db47 in redmond tenant and 975f013f-7f24-47e8-a7d3-abc4752bf346 in pme

**Review by chkeita — APPROVED** [review]

File: src/deployment/deploy.py [hunk 2]
```diff
@@ -427,6 +427,14 @@ def try_sp_create() -> None:
...
+                "authority": authority,
+            }
```
[the remark this answers is not in the document]
Comment (muhammadmeisam) [reply src/deployment/deploy.py]: Done

File: src/deployment/deploy.py [hunk 3]
```diff
@@ -426,7 +426,16 @@ def try_sp_create() -> None:
             }
 
         else:
+            onefuzz_cli_app = cli_app[0].app_id
             authorize_application(uuid.UUID(ONEFUZZ_CLI_APP), app.app_id)
```
Comment (chkeita) [inline src/deployment/deploy.py:430]: my bad there was a error in the code suggested
```suggestion 
onefuzz_cli_app = cli_app[0]
authorize_application(uuid.UUID(onefuzz_cli_app.app_id), app.app_id)
```

**Review by bmc-msft — CHANGES_REQUESTED** [review]
When deploying this PR, I get this error:
```
INFO:deploy:checking if RBAC already exists
INFO:deploy:creating Application registration
INFO:deploy:creating service principal
Traceback (most recent call last):
  File "deploy.py", line 1077, in <module>
    main()
  File "deploy.py", line 1071, in main
    state[1](client)
  File "deploy.py", line 434, in setup_rbac
    authority = onefuzz_cli_app.authority
AttributeError: 'Application' object has no attribute 'authority'
```

File: src/deployment/deploy.py [hunk 4]
```diff
@@ -426,7 +427,17 @@ def try_sp_create() -> None:
...
 
         else:
-            authorize_application(uuid.UUID(ONEFUZZ_CLI_APP), app.app_id)
+            onefuzz_cli_app = cli_app[0]
+            authorize_application(uuid.UUID(onefuzz_cli_app.app_id), app.app_id)
+            if self.multi_tenant_domain:
+                authority = COMMON_AUTHORITY
+            else:
+                onefuzz_client = get_graph_client(self.get_subscription_id())
+                authority = "https://login.microsoftonline.com/%s" % onefuzz_client.config.tenant_id
```
Comment (bmc-msft) [inline src/deployment/deploy.py:436]:
```suggestion
                authority = (
                    "https://login.microsoftonline.com/%s"
                    % onefuzz_client.config.tenant_id
                )
```

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**